pkg/controller/servicebinding/secret.go

```diff
@@ -27,10 +32,68 @@ func (s *secret) buildResourceClient() dynamic.ResourceInterface {
 	return s.client.Resource(gvr).Namespace(s.ns)
 }
 
-// createOrUpdate will take informed payload and either create a new secret or update an existing
-// one. It can return error when Kubernetes client does.
-func (s *secret) createOrUpdate(payload map[string][]byte, ownerReference metav1.OwnerReference) (*unstructured.Unstructured, error) {
+// compare existing secret and new secret
+func (s *secret) isSame(secretName string, payload map[string][]byte) bool {
 	logger := s.logger.WithValues("Namespace", s.ns, "Name", s.name)
+
+	resourceClient := s.buildResourceClient()
+	existingSecret, err := resourceClient.Get(secretName, metav1.GetOptions{})
+	if err != nil {
+		logger.Error(err, "Error fetching secret")
+		return false
+	}
+	existingSecretData, _, _ := unstructured.NestedMap(existingSecret.Object, "data")
+
+	payloadInterim := make(map[string]interface{})
+	for k, v := range payload {
+		payloadInterim[k] = base64.StdEncoding.EncodeToString(v)
+	}
+
+	comparisonResult := nestedMapComparison(existingSecretData, payloadInterim)
+
+	if comparisonResult.Success {
+		logger.Debug("Secret data is same.")
+		return true
+	}
+
+	return false
+}
+
+func sortKeys(payload map[string][]byte) string {
+	keys := make([]string, 0, len(payload))
+	for k := range payload {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	var text []string
+	for _, k := range keys {
+		s := fmt.Sprintf("%s%s", k, string(payload[k]))
+		text = append(text, s)
+	}
+	return strings.Join(text, "")
+}
+
+func hash(text string) (string, error) {
+	algorithm := sha1.New()
+	_, err := algorithm.Write([]byte(text))
+	if err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(algorithm.Sum(nil)), nil
+}
+
+// create will take informed payload and create a new secret
+// It can return error when Kubernetes client does.
+func (s *secret) create(payload map[string][]byte, ownerReference metav1.OwnerReference) (*unstructured.Unstructured, error) {
+	logger := s.logger.WithValues("Namespace", s.ns, "Name", s.name)
+
+	uniqueSuffix, err := hash(sortKeys(payload))
+	if err != nil {
+		return nil, err
+	}
+	s.name = s.name + "-" + uniqueSuffix
```

Comment:
An utility function can be extracted with this bureaucracy:

```go
func uniqueSuffix(name string, payload map[string][]byte) string { ... }
```

This could be then tested and used within tests to verify client and operator behavior (as in "is the operator naming properly the service binding secret?")

Reply:
Done here 4302e7f